Userland/Libraries/LibC/unistd.cpp

```diff
@@ -975,8 +975,41 @@ int unveil(char const* path, char const* permissions)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpass.html
 char* getpass(char const* prompt)
 {
-    dbgln("FIXME: getpass('{}')", prompt);
-    TODO();
+    int tty = open("/dev/tty", O_RDWR | O_NOCTTY | O_CLOEXEC);
+
+    if (tty < 0)
+        return nullptr;
+
+    struct termios backup, noecho;
+    tcgetattr(tty, &backup);
+
+    noecho = backup;
+    noecho.c_lflag &= ~(ECHO|ISIG|INLCR|IGNCR);
+    noecho.c_lflag |= ICANON | ICRNL;
+
+    tcsetattr(tty, TCSAFLUSH, &noecho);
```

Comment:
tcsetattr can return error codes, and this is not handled here.  This is a pretty sensitive function to skip checking errors, IMO.